tests/Engine/EngineTests/EngineDataTest.php

```diff
@@ -1,69 +1,56 @@
 <?php
 
+namespace FlagsmithTest\Engine\EngineTests;
+
 use Flagsmith\Engine\Engine;
-use Flagsmith\Engine\Environments\EnvironmentModel;
-use Flagsmith\Engine\Identities\IdentityModel;
+use Flagsmith\Engine\Utils\Types\Context\EvaluationContext;
 use PHPUnit\Framework\TestCase;
 
 class EngineDataTest extends TestCase
 {
     private int $attempt = 0;
+
+    /** @return array<array<mixed>> */
     public function extractTestCases()
     {
-        $fileContents = file_get_contents(__DIR__ . '/EngineTestData/data/environment_n9fbf9h3v4fFgH3U3ngWhb.json');
-
-        $contents = json_decode($fileContents);
-
-        $environmentModel = EnvironmentModel::build($contents->environment);
+        $testDataContent = file_get_contents(
+            __DIR__ .
+                '/EngineTestData/data/environment_n9fbf9h3v4fFgH3U3ngWhb.json',
+        );
+        $testData = json_decode($testDataContent, associative: false);
 
         $parameters = [];
-        foreach ($contents->identities_and_responses as $testCase) {
-            $parameters[] = [
-                $environmentModel,
-                IdentityModel::build($testCase->identity),
-                $testCase->response
-            ];
+        foreach ($testData->test_cases as $testCase) {
+            $context = EvaluationContext::fromJsonObject($testCase->context);
+            $parameters[] = [$context, $testCase->result];
         }
 
         return $parameters;
     }
 
     /**
      * @dataProvider extractTestCases
+     * @param EvaluationContext $evaluationContext
+     * @param object $expectedEvaluationResult
+     * @return void
      */
-    public function testEngine($environmentModel, $identityModel, $expectedResponse)
-    {
-        $this->attempt++;
-        $engineResponse = Engine::getIdentityFeatureStates($environmentModel, $identityModel);
-
-        usort(
-            $engineResponse,
-            fn ($fs1, $fs2) => $fs1->getFeature()->getName() <=> $fs2->getFeature()->getName()
-        );
-
-        $flags = $expectedResponse->flags;
-        usort(
-            $flags,
-            fn ($fs1, $fs2) => $fs1->feature->name <=> $fs2->feature->name
-        );
-
+    public function testEngine(
+        $evaluationContext,
+        $expectedEvaluationResult,
+    ): void {
+        // When
+        $evaluationResult = Engine::getEvaluationResult($evaluationContext);
+
+        // Hack to allow comparing flags as associative arrays (<feature_name>: <flag>)
+        $wanted = array_column($expectedEvaluationResult->flags, null, 'name');
+        $expectedEvaluationResult->flags = $wanted;
+        $actual = array_column($evaluationResult->flags, null, 'name');
+        $evaluationResult->flags = $actual;
```

Comment:
> [!NOTE]
> I would prefer `EvaluationResult->flags` to be an associative array `<feature_name> => <flag>`, rather than an indexed array. This would improve data consistency and avoid this hack. cc @khvn26

Reply:
Addressed in https://github.com/Flagsmith/engine-test-data/pull/12.